src/main/java/org/lisoft/lsml/model/item/BallisticWeapon.java

```diff
@@ -73,7 +73,7 @@ public BallisticWeapon(
               // HeatSource Arguments
               aHeat,
               // Weapon Arguments
-              aCooldown, aRangeProfile, aRoundsPerShot, aDamagePerProjectile, aProjectilesPerRound, aProjectileSpeed,
+              aCooldown, aRangeProfile, aRoundsPerShot, aRoundsPerShot, aDamagePerProjectile, aProjectilesPerRound, aProjectileSpeed,
```

Comment:
I'm a bit concerned that passing `aRoundsPerShot` to volleysize is not correct for all weapons. E.g. C-UAC/20 has 3 rounds per shot but a volley size of 1. Could you add tests on C-UAC/20 and (C-)LB-10X?

Reply:
aRoundsPerShot is not equal to aProjectilesPerRound.   lbx10 fires 10 projectiles per round but only 1 round per shot.  CUAC20 fires 4 projectiles per round but only 1 round per shot.  'aRoundsPerShot' in LSML is 'ammoPerShot' in the PGI file.  'aProjectilesPerRound' in LSML is 'numFiring' in the PGI file.  

the ballistic weapons do not have the 'volleysize' attribute in the PGI file. so I chose aRoundsPerShot over aProjectilesPerRound.  I may be incorrect I will do more testing in the game.

all LSML AmmoWeapons have the 'volleydelay' attribute in the PGI file.

the file is MechWarrior Online\Game\GameData.pak\Libs\Items\Weapons\Weapons.xml.

Reply:
Well, did more testing.  While it is true that the additional trigger pull on the UAC does not impact the overall cooldown (other than when it jams), the initial volley delay does in fact get added into the firing time.  

so, that really means I should have used aProjectilesPerRound to set volleySize.  

UACs have a volleyDelay of num>0.  LBx has volleyDelay of 0. 

But, the calc for UAC weapons rFP and eFP will be impacted.

Reply:
ok, I have corrected the code to convert the ballistic projectiles per shot into something that can be used to determine firing delay.  I believe this is now finally correctly calculated for all types of weapons.  I have added tests for LBX and CUAC

Reply:
That was far worse than I thought.  So I rebuilt a bunch of the setting of volleySize and the creation of a firingDelay variable set at AmmoWeapon initiation.  I dont know of any quirks that shorten that but it would be easy to add if one does exist.

anyway.  I feel pretty good about the firing delay calculation for all weapon types now.. please dig in to see if you find anything I missed.

Reply:
I think this is correct now, bar my separate comment about `ammoPerShot`. I would have passed volley size = 1 through the constructor from the parsing code but this is realistically going to be correct for all ballistic weapons and is heading into YAGNI territory.

Obligatory:
![monkeyuser.com](https://res.cloudinary.com/practicaldev/image/fetch/s--u53Aunfr--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/tq3ggyimlj9h32ektgrx.png)